### ISSUE TITLE
Virtual Keyboard Support

### ISSUE THREAD

**Issue by TheoRettisch** [M4rtinK/modrana#269]
Hello,

i would like to use Modrana on a linux system with touchscreen input. Everything is working very well with the awesome QT5 UI, only thing i am missing is an on screen keyboard for the poi and address search fields. There seems to be the Qt Virtual Keyboard which can handle exactly this. Any chance to include it into Modrana and maybe make it an startup option?

Btw. would it be possible to always set the start point of an route to the current location?

**Comment by M4rtinK** [comment]
> Hello,
> 
> i would like to use Modrana on a linux system with touchscreen input. Everything is working very well with the awesome QT5 UI, only thing i am missing is an on screen keyboard for the poi and address search fields. There seems to be the Qt Virtual Keyboard which can handle exactly this. Any chance to include it into Modrana and maybe make it an startup option?

Oh, so you environment is basically a fairly standard desktop distro running on a touchscreen only device ? Then indeed, input could be an issue. On "mobile" platforms such as Sailfish OS or Android,  this somehow works auto-magically - when an input field is clicked a system provided virtual keyboard shows up. Unfortunately I don't really know how this works & if it could be adapted for your case.

In such case indeed Qt Virtual Keyboard might be a solution, even though it's documentation and especially usage examples seem to be kinda lacking, at least after a brief search.

> 
> Btw. would it be possible to always set the start point of an route to the current location?
Do you mean when using the on-screen routing buttons ?

These are actually more or less temporary, eventually I would like to introduce a full featured routing page where it will be possible to set start, destination & waypoints  (and other route parameters) from various sources, including current position.

On the other hand, there are various routing options already available that always route from current position:
- long lick on a point on the map, from the menu that shows up select "route here"
- routing to a search result
- routing to a POI

**Comment by TheoRettisch** [comment]
> Oh, so you environment is basically a fairly standard desktop distro running on a touchscreen only device ? Then indeed, input could be an issue. On "mobile" platforms such as Sailfish OS or Android, this somehow works auto-magically - when an input field is clicked a system provided virtual keyboard shows up. Unfortunately I don't really know how this works & if it could be adapted for your case.

Yes, i am using openbox window manager with modrana running fullscreen on one virtual desktop. Apart from the Qt Virtual Keyboard there seems to also exist this project which might help linux users:

[https://wiki.gnome.org/Projects/Caribou](url)

Activation of the keyboard is handled via dbus events. This might be much more coding though, The Qt method sounded to me like very easy to implement. But so far i am already failing at the include statement when trying it myself.

> long lick on a point on the map, from the menu that shows up select "route here"

That is exactly what i was looking for, Thanks!
